### ISSUE TITLE
Proof of concept for lark

### ISSUE THREAD

**Issue by r-e-stern** [uchicago-cs/chiventure#862]
Attempt to write some basic code in `lark`, a Python parsing library, in order to assess whether we should use it to write the DSL.

**Comment by jacknugent1529** [comment]
I created a grammar and program in lark to parse wdl files, which is a subset of yaml. Lark uses EBNF grammar, which is similar to most parsing libraries we have looked at. Lark uses one grammar for both the parser and lexer, which means the code is shorter and easier to write. Additionally, the grammar for lark can be written in a separate file from the functions which manipulate the parsing tree, and this separation makes it easier to manipulate one specific part of the parsing program. The grammar I used for the wdl files is shown below.
```
//wdl.lark

?start: _NL dict?

dict: pair+

pair: key ":" value

key: string -> string

value: string _NL -> string
     | _NL _INDENT bullet_point+ _DEDENT -> list

bullet_point: "-" pair [_INDENT pair+ _DEDENT]? -> dict


string: ESCAPED_STRING -> escaped_string
      | STRING -> string

%import common.CNAME -> STRING
%import common.WS_INLINE
%declare _INDENT _DEDENT
%ignore WS_INLINE

%import common.ESCAPED_STRING

COMMENT: "#" /[^\n]/*
%ignore COMMENT

_NL: ( /(\r?\n[\t ]*)/ | COMMENT) +
```

**Comment by r-e-stern** [comment]
I also scoped out Lark this week, following along with the [tutorial from its documentation](https://github.com/lark-parser/lark/blob/master/docs/json_tutorial.md) on how to build a [JSON parser](https://github.com/lark-parser/lark/blob/master/examples/json_parser.py), which turned out to have very similar features to Jack's WDL parser, since the two languages share a number of features. I got it working on the CS Linux machines with very little difficulty. @jacknugent1529 mentioned that he heard in office hours that WDL is undergoing a transition to JSON, so the code for the lexer from the JSON parser could turn out to be useful to us in the future.

**Pros**
- Lark allows you to concisely specify a grammar, in roughly the same way that ANTLR (#822) does.
- Lark features built-in handling of indentation, escaped strings, and other useful patterns.
- Lark is a well-maintained, widely-used, computationally-efficient (at least, per its claims) Python module.
- Lark supports a variety of parsing algorithms, which is something that sounds like it would be important but I have no idea if it actually is.
- Both the specification of a grammar and the implementation of a lexer in Lark feel, well, not necessarily _intuitive_, but as close to it as you're gonna get.

**Cons**
- Parsers are hard, man.
- The most common gripes people have about Lark are its [lack of verbose error messages](https://github.com/lark-parser/lark/issues/480#issuecomment-623765761).

As a group, we met to discuss all the Python parsers we looked at this week (#865), and decided to use Lark.
